File: problems/2104_sum_of_subarray_ranges/Solution.java
class Solution {
    public long subArrayRanges(int[] nums) {
        long result = 0;
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            int minVal = nums[i], maxVal = nums[i];
            for (int j = i + 1; j < n; j++) {
                minVal = Math.min(minVal, nums[j]);
                maxVal = Math.max(maxVal, nums[j]);
                result += maxVal - minVal;
            }
        }
        return result;
    }
}

File: problems/2104_sum_of_subarray_ranges/solution.py
class Solution:
    def subArrayRanges(self, nums: list[int]) -> int:
        result = 0
        n = len(nums)
        for i in range(n):
            min_val = max_val = nums[i]
            for j in range(i + 1, n):
                min_val = min(min_val, nums[j])
                max_val = max(max_val, nums[j])
                result += max_val - min_val
        return result
